scripts/oneoff/generate-g-cloud-services.py

```diff
@@ -0,0 +1,163 @@
+    while services_generated < args.quantity:
+        random_lot = random.choice(args.lots)
+        random_supplier = random.choice(suppliers)
+
+        supplier_id = random_supplier['supplierId']
+        if supplier_id not in suppliers_prepared:
+            # 2a) For each supplier, register interest in a given (new) framework
+            data_api_client.register_framework_interest(supplier_id=supplier_id,
+                                                        framework_slug=args.new_slug,
+                                                        user=identity)
+
+            # 2b) Submit and mark as complete a valid supplier declaration for the new framework
+            data_api_client.set_supplier_declaration(supplier_id=supplier_id,
+                                                     framework_slug=args.new_slug,
+                                                     declaration=eligible_g9_declaration_base(),
+                                                     user=identity)
+
+        fake_service = gcloud_service_faker.generate_from_file(filepath_service_lot_schema(args.new_slug, random_lot))
+        fake_service['serviceName'] = gen_gcloud_name()
+
+        draft_service = data_api_client.create_new_draft_service(framework_slug=args.new_slug,
+                                                                 lot=random_lot,
+                                                                 supplier_id=supplier_id,
+                                                                 data=fake_service,
+                                                                 user=identity)
+
+        data_api_client.complete_draft_service(draft_id=draft_service['services']['id'],
+                                               user=identity)
+
+        services_generated += 1
+
+    # 4) Set a given (new) framework to standstill
+    data_api_client._post_with_updated_by(url='{}/frameworks/{}'.format(data_api_url, args.new_slug),
+                                          data={"frameworks": {"status": "standstill"}},
+                                          user=identity)
+
+    # a) Runs the `mark-definite-framework-results.py` script to mark x suppliers as on the new framework.
+    subprocess.call(['python', 'scripts/mark-definite-framework-results.py', '--reassess-failed-sf', '--updated-by',
+                     identity, args.env, args.data_api_token, args.new_slug, filepath_declaration_validator])
+
+    # b) Runs the `make-g-cloud-live.py` script (ignoring all documents)
+    subprocess.call(['python', 'scripts/make-g-cloud-live.py', '--ignore-documents', args.new_slug, args.env,
+                     args.data_api_token, 'null', 'null'])
+
+    # c) Mark framework agreements as returned for all selected suppliers
+    # TODO: Maybe do this. Not doing it breaks some supplier summary pages, but shouldn't affect browsing the catalogue.
+
+    # 5) Set a given (new) framework to live
```

Comment:
To be honest I think it's a bit weird having a script flipping framework statuses at all.  This one needs to do it so we can submit drafts and publish services.  But the state of frameworks should generally depend on what we as a team need them to be for testing things (and in some cases what the functional tests need them to be for testing things).

For example, if this was any previous framework application window we would have functional tests running and submitting services in each lot every 30 minutes.  Flipping the framework status to 'live' would then cause those tests to all fail.  We would need to agree as a team what status the live environment should currently be in - just because a script has run isn't really a reason for moving framework status on.

Thinking about it, there's an argument that the script should probably re-set the framework status to whatever it was to begin with.